includes/customers/customer-actions.php

```diff
@@ -11,6 +11,124 @@
  * @since       3.0
  */
 
+/**
+ * Is intended to be used when adding a customer directly with edd_add_customer function.
+ *
+ * Because the edd_add_customer function only directly interacts with the customers DB table, we may
+ * need to do some additional actions like adding items to the customer email addresses table.
+ *
+ * @since 3.0.4
+ */
+function edd_process_customer_added( $customer_id, $data ) {
+	// Make sure we add a new primary email address to the email addresses table.
+	edd_add_customer_email_address(
+		array(
+			'customer_id' => $customer_id,
+			'email'       => $data['email'],
+			'type'        => 'primary',
+		)
+	);
+}
+add_action( 'edd_customer_added', 'edd_process_customer_added', 10, 2 );
+
+/**
+ * Is intended to be used when updating a customer directly with edd_add_customer function.
+ *
+ * Because the edd_update_customer function only directly interacts with the customers DB table, we may
+ * need to do some additional actions like managing email addresses.
+ *
+ * @since 3.0.4
+ */
+function edd_process_customer_updated( $customer_id, $data, $previous_customer_data ) {
```

Comment:
I would really love to see the parameters documented here. I also think I would like for `$previous_customer_data` to be renamed to something that feels more like an object since it is actually the previous customer object, rather than an array of data, and we tend to name things `_data` when they're arrays.

Reply:
Fixed in https://github.com/awesomemotive/easy-digital-downloads/pull/9396/commits/1d7a8828ec86ed0eb6151b6ea1da33e1ae59f227